src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -68,278 +69,462 @@ def __init__(self):
-                return
-            self.kafka_connection.publish(results, self._simulation_id)
-
-    def _update_progress_info(self, slot_no, slot_count):
-        run_duration = (
-                now(tz=TIME_ZONE) - self.run_start -
-                duration(seconds=self.paused_time)
-        )
-
-        if gsy_e.constants.RUN_IN_REALTIME:
-            self.progress_info.eta = None
-            self.progress_info.percentage_completed = 0.0
-        else:
-            self.progress_info.eta = (run_duration / (slot_no + 1) * slot_count) - run_duration
-            self.progress_info.percentage_completed = (slot_no + 1) / slot_count * 100
-
-        self.progress_info.elapsed_time = run_duration
-        self.progress_info.current_slot_str = get_market_slot_time_str(
-            slot_no, self.simulation_config)
-        self.progress_info.next_slot_str = get_market_slot_time_str(
-            slot_no + 1, self.simulation_config)
-        self.progress_info.current_slot_number = slot_no
-
-    def set_area_current_tick(self, area, current_tick):
-        area.current_tick = current_tick
-        for child in area.children:
-            self.set_area_current_tick(child, current_tick)
-
     def _get_current_market_time_slot(self, slot_number: int) -> DateTime:
         return (self.area.config.start_date + (slot_number * self.area.config.slot_length)
                 if GlobalConfig.IS_CANARY_NETWORK else self.area.now)
 
-    def _calculate_total_initial_ticks_slots(self, config, slot_resume, tick_resume):
-        slot_count = int(config.sim_duration / config.slot_length)
-
-        if gsy_e.constants.RUN_IN_REALTIME:
-            slot_count = sys.maxsize
-
-            today = datetime.date.today()
-            seconds_since_midnight = time() - mktime(today.timetuple())
-            slot_resume = int(seconds_since_midnight // config.slot_length.seconds) + 1
-            seconds_elapsed_in_slot = seconds_since_midnight % config.slot_length.seconds
-            ticks_elapsed_in_slot = seconds_elapsed_in_slot // config.tick_length.seconds
-            tick_resume = int(ticks_elapsed_in_slot) + 1
-
-            seconds_elapsed_in_tick = seconds_elapsed_in_slot % config.tick_length.seconds
-
-            seconds_until_next_tick = config.tick_length.seconds - seconds_elapsed_in_tick
-
-            ticks_since_midnight = int(seconds_since_midnight // config.tick_length.seconds) + 1
-            self.set_area_current_tick(self.area, ticks_since_midnight)
-
-            sleep(seconds_until_next_tick)
-
-        if self.slot_length_realtime:
-            self.tick_length_realtime_s = self.slot_length_realtime.seconds / \
-                                          self.simulation_config.ticks_per_slot
-        return slot_count, slot_resume, tick_resume
-
     def _execute_simulation(self, slot_resume, tick_resume, console=None):
-        self.current_expected_tick_time = self.run_start
         config = self.simulation_config
```

Comment:
I stumble over this a lot. How about taking the opportunity and either use `config` in the whole method or (my preference) use `self. simulation_config` only. What do you think?

Reply:
Let me actually do it. I will self-contain it in one commit and hopefully it will not be that much of a fuss